File: app/src/main/java/com/mxt/anitrend/view/activity/detail/NotificationActivity.java
package com.mxt.anitrend.view.activity.detail;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.mxt.anitrend.R;
import com.mxt.anitrend.base.custom.activity.ActivityBase;
import com.mxt.anitrend.databinding.ActivityFrameGenericBinding;
import com.mxt.anitrend.presenter.base.BasePresenter;
import com.mxt.anitrend.view.activity.index.MainActivity;
import com.mxt.anitrend.view.fragment.detail.NotificationFragment;

/**
 * Created by max on 2017/10/25.
 */

public class NotificationActivity extends ActivityBase<Void, BasePresenter> {

    private ActivityFrameGenericBinding binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityFrameGenericBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        mSearchView = binding.customToolbar.searchView;
        setSupportActionBar(binding.customToolbar.toolbar);
    }

    @Override
    protected void onPostCreate(@Nullable Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        setPresenter(new BasePresenter(this));
        onActivityReady();
    }

    @Override
    public void onBackPressed() {
        if (isTaskRoot())
            startActivity(new Intent(this, MainActivity.class));

        super.onBackPressed();
    }

    /**
     * Make decisions, check for permissions or fire background threads from this method
     * N.B. Must be called after onPostCreate
     */
    @Override
    protected void onActivityReady() {
        mFragment = NotificationFragment.Companion.newInstance();
        updateUI();
    }

    @Override
    protected void updateUI() {
        if(mFragment != null) {
            FragmentManager fragmentManager = getSupportFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.replace(R.id.content_frame, mFragment, mFragment.TAG);
            fragmentTransaction.commit();
        }
    }

    @Override
    protected void makeRequest() {

    }
}

File: app/src/main/java/com/mxt/anitrend/view/activity/detail/NotificationActivity.kt
package com.mxt.anitrend.view.activity.detail

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import com.mxt.anitrend.R
import com.mxt.anitrend.databinding.ActivityFrameGenericBinding
import com.mxt.anitrend.view.activity.CommonActivity
import com.mxt.anitrend.view.activity.index.MainActivity
import com.mxt.anitrend.view.fragment.detail.NotificationFragment

class NotificationActivity : CommonActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = ActivityFrameGenericBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.customToolbar.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val fragment = NotificationFragment.newInstance()
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.content_frame, fragment, fragment.TAG)
            .commit()
    }

    @Deprecated("Deprecated in Java")
    override fun onBackPressed() {
        super.onBackPressed()
        navigateBack()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            navigateBack()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    @Suppress("DEPRECATION")
    private fun navigateBack() {
        if (isTaskRoot) {
            startActivity(Intent(this, MainActivity::class.java))
        }
        super.onBackPressed()
    }
}
